sourcefiles/elementrando.py

```diff
@@ -0,0 +1,550 @@
+
+        for i, elem in enumerate(elems):
+            real_i = i
+            if i == 3:
+                real_i = 4
+            elif i == 4:
+                real_i = 6
+            statblock = config.pcstats.pc_stat_dict[C(real_i)].stat_block
+            statblock.innate_element_resisted = elem
+
+        config.elems = elems + roboelems
+        config.tech_db = shuffle_techdb(config.tech_db, elems, roboelems)
+
+def update_ctrom(ct_rom: ctrom.CTRom, config: cfg.RandoConfig):
+    update_scripts(ct_rom, config)
+
+def update_scripts(ct_rom: ctrom.CTRom, config: cfg.RandoConfig):
+    if len(config.elems) > 0:
+        spekkio_script = ct_rom.script_manager.get_script(L.SPEKKIO)
+
+        for i, barr in enumerate(spekkio_script.strings):
+            string = ctstrings.CTString(barr).to_ascii() 
+            newstr = None
+            if re.search(r'punk hairdo', string) is not None: # Crono
+                newstr = re.sub('Lightning', str(config.elems[0]), string)
+            elif re.search(r'ponytail', string) is not None: # Marle
+                newstr = re.sub('Water', str(config.elems[1]), string)
+            elif re.search(r'goofy glasses', string) is not None: # Lucca
+                newstr = re.sub('Fire', str(config.elems[2]), string)
+            elif re.search(r'biggest toy', string) is not None: # Robo
+                newstr = 'SPEKKIO: That\'s the biggest toy I\'ve{line break}ever seen...{line break}Hey, you\'re not alive, are you?!{page break}You\'ve got great strength, however,{line break}since I can\'t measure your inner{line break}character, I can\'t give any magic to{line break}you.{page break}But your various weapons will suffice.{line break}They can inflict many types{line break}of damage.{null}'
+            elif re.search(r'a frog', string) is not None: # Frog
+                newstr = re.sub('Water', str(config.elems[3]), string)
+            elif re.search(r'Sweetheart', string) is not None: # Ayla
+                pass # no changes for ayla
+            elif re.search(r'marlin', string) is not None: # Magus
+                newstr = re.sub('Shadow', str(config.elems[4]), string)
+
+            if newstr is not None:
+                spekkio_script.strings[i] = ctstrings.CTString.from_str(newstr)
+                spekkio_script.modified_strings = True
+
+def setelem(tech, elem: El):
+    tech['control'][3] &= 0x0F
+    if elem == El.LIGHTNING:
+        tech['control'][3] |= 0x80
+    elif elem == El.SHADOW:
+        tech['control'][3] |= 0x40
+    elif elem == El.ICE:
+        tech['control'][3] |= 0x20
+    elif elem == El.FIRE:
+        tech['control'][3] |= 0x10
+    return tech
+
+_common = {
+        'fire': '*Fire',
+        'ice': '*Ice',
+        'lit': '*Lightning',
+        'fire2': '*Fire 2',
+        'ice2': '*Ice 2',
+        'lit2': '*Lightning2',
+        'dmist': '*Dark Mist',
+}
+
+# TechID: {Element: {'name': 'NewName', 'gfx': {4: 0x63, 5: 0x98 etc.}}}
+_replacements: Dict = {
```

Comment:
This is a little bit ugly but I'm not really sure if there's a better way. At best, maybe there's a way to write is as more of a table instead, but I'm unsure about requiring some sort of parsing.